pkgs/development/python-modules/paho-mqtt/default.nix

```diff
@@ -1,42 +1,58 @@
 { lib
 , stdenv
 , buildPythonPackage
+, pythonOlder
 , fetchFromGitHub
-, isPy3k
+, hatchling
 , pytestCheckHook
-, mock
-, six
 }:
 
-buildPythonPackage rec {
+let
+  testing = fetchFromGitHub {
+    owner = "eclipse";
+    repo = "paho.mqtt.testing";
+    rev = "9d7bb80bb8b9d9cfc0b52f8cb4c1916401281103";
+    hash = "sha256-3QqCLgV4WMZuG8+3nIa3g9PtzH7cXjw1fWbZkj7H3RU=";
+  };
+in buildPythonPackage rec {
   pname = "paho-mqtt";
-  version = "1.6.1";
-  format = "setuptools";
+  version = "2.0.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "eclipse";
     repo = "paho.mqtt.python";
     rev = "v${version}";
-    hash = "sha256-9nH6xROVpmI+iTKXfwv2Ar1PAmWbEunI3HO0pZyK6Rg=";
+    hash = "sha256-dR/MCz3c9eHai76I17PGD71E5/nZVEo6uRwUULOzKQU=";
   };
 
+  build-system = [
+    hatchling
+  ];
+
   nativeCheckInputs = [
     pytestCheckHook
-    six
-  ] ++ lib.optionals (!isPy3k) [
-    mock
   ];
 
   doCheck = !stdenv.isDarwin;
 
+  preCheck = ''
+    ln -s ${testing} paho.mqtt.testing
+
+    export PYTHONPATH=".:$PYTHONPATH"
```

Comment:
Hold on, isn't this going to cause it to pick up the copy of `paho-mqtt` in the source directory instead of the installed copy?